githubutils/repo.go

```diff
@@ -145,17 +145,100 @@ func GetRawGitFile(ctx context.Context, client *github.Client, content *github.R
 	return byt, err
 }
 
+// A RepoFilter can check if a release shold be filtered out based on a condition.
+//  In general filters may be stateful and be a StatefulRepoFilter
+type RepoFilter interface {
+	// Apply the filter and return true if it conforms to the flter and should
+	// not be removed.
+	Apply(release *github.RepositoryRelease) bool
+}
+
+// A StatefulRepoFilter has a stateful prerun check and along with a way to determine
+// if a release conforms to our desired filter.
+type StatefulRepoFilter interface {
```

Comment:
I don't fully understand how a StatefulFilter works. To me it seems like the stateful filter PreFilterCheck is a way of defining the state on a Filter (to be used during the filter logic). 
I wonder if instead, we just stateful component should just be handled at construction time. That way, if we create a RepoFilter (Or RepositoryReleasePredicate), we can instantiate it to behave a certain way (ie filter only latest release), and then the code doesn't need to be aware that it has any internal state.

Reply:
So this is the crux of the issue (other than the related issue where in issue updates we retrun too early which breaks things)

Namely we only get releases in go-utils today.
So if we want to only upload issues for the most recent LTS branch then we need to be able to determine that either here or  leak repository retrieval info to another library. This is true because one can only know what an LTS branch is by inspection of the other existing releases.